platform/ios/src/MGLMapView.mm

```diff
@@ -695,8 +698,10 @@ - (void)dealloc
     [self.attributionButtonConstraints removeAllObjects];
     self.attributionButtonConstraints = nil;
     
+    [self.locationManager stopUpdatingLocation];
```

Comment:
The accesses via `self` in `dealloc` should probably be replaced by the instance variable, so `[_locationManager stopUpdatingLocation]` etc.

Reply:
What would be the reason for this?

Reply:
It's more a rule of thumb; in `init` and `dealloc` the instance is in the process of being constructed or destroyed and as such can be in an intermediate state. We could have a custom accessor (e.g. for `locationManager` in this case) that does something that assumes a certain state.